lib/src/modules/status.cc

```diff
@@ -21,8 +26,10 @@ namespace Modules {
 namespace fs = boost::filesystem;
 namespace lth_file = leatherman::file_util;
 namespace HW = HorseWhisperer;
+namespace pcp_util = PCPClient::Util;
 
 static const std::string QUERY { "query" };
+static const uint32_t METADATA_RACE_MS { 100 };
```

Comment:
The race @nicklewis talks about in PCP-168 is due to a very short time window, where the non-blocking process terminates, leatherman's execute() returns, and the non-blocking action task updates the metadata file. If a status request is processed during such interval, the PID exists() function will return false, but metadata.completed would be (wrongly) false. In that case (PID does not exist and metadata still not updated) the above delay will allow the non-blocking task to acquire the transaction mutex and update the metadata before the status handler reads it. The duration value was chosen by considering that 1) the race window is very tight and 2) 100 ms is an order of magnitude less compared to the status polling period of the clj-pxp-puppet function (it shouldn't impact the overall transaction).